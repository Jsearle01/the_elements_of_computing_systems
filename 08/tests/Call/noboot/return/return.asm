@LCL
D=M
@R13
M=D
@R14
M=D
@5
D=A
@R14
M=M-D
A=M
D=M
@R14
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
D=M
@SP
M=D
M=M+1
@R13
M=M-1
A=M
D=M
@THAT
M=D
@R13
M=M-1
A=M
D=M
@THIS
M=D
@R13
M=M-1
A=M
D=M
@ARG
M=D
@R13
M=M-1
A=M
D=M
@LCL
M=D
@R14
A=M
0;JMP
