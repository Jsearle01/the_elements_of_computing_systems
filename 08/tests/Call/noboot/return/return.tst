load return.asm,
output-file return.out,
compare-to return.cmp,
output-list
    RAM[0]%D1.6.1
    RAM[1]%D1.6.1
    RAM[2]%D1.6.1
    RAM[256]%D1.6.1
    RAM[257]%D1.6.1
    RAM[258]%D1.6.1
    RAM[259]%D1.6.1
    RAM[260]%D1.6.1
    RAM[261]%D1.6.1;

set RAM[0] 261,
set RAM[1] 261,
set RAM[2] 256,

set RAM[256] 200,
set RAM[257] 101,
set RAM[258] 102,
set RAM[259] 103,
set RAM[260] 104,

repeat 100 {
  ticktock;
}

output;

