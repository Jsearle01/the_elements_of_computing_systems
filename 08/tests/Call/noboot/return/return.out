| RAM[0] | RAM[1] | RAM[2] |RAM[256]|RAM[257]|RAM[258]|RAM[259]|RAM[260]|RAM[261]|
|    257 |    101 |    102 |     69 |    101 |    102 |    103 |     69 |      0 |
